platine-management-service/src/main/java/fr/insee/survey/datacollectionmanagement/questioning/service/impl/QuestioningAccreditationServiceImpl.java

```diff
@@ -39,9 +37,25 @@ public QuestioningAccreditation saveQuestioningAccreditation(QuestioningAccredit
         return questioningAccreditationRepository.save(questioningAccreditation);
     }
 
+    @Override
+    public List<QuestioningAccreditation> findByQuestioningIdAndIsMain(Long questioningId) {
+        return questioningAccreditationRepository.findAccreditationsByQuestioningIdAndIsMainTrue(questioningId);
+    }
+
     @Override
     public void deleteAccreditation(QuestioningAccreditation acc) {
         questioningAccreditationRepository.deleteById(acc.getId());
     }
 
+    @Override
+    public void setMainQuestioningAccreditationToContactAsMain(String contactId, Long questioningId) {
+
+        List<QuestioningAccreditation> questioningAccreditations = findByQuestioningIdAndIsMain(questioningId);
+
+        for(QuestioningAccreditation qa :  questioningAccreditations){
+            qa.setIdContact(contactId);
+            saveQuestioningAccreditation(qa);
+        }
+    }
```

Comment:
there should be only one main accreditation => could use Optional<QA> instead of List<QA>